codecarbon/lock.py

```diff
@@ -0,0 +1,60 @@
+"""
+Ensures that only one instance of codecarbon is running at a time.
+It creates a lock file in /tmp/.codecarbon.lock and removes it on exit.
+If the lock file already exists, it exits the program.
+"""
+
+import atexit
+import errno
+import os
+import sys
+import tempfile
+
+from codecarbon.external.logger import logger
+
+# We use tempfile.gettempdir() to get the system's temporary directory (linux: /tmp, windows: C:\Users\username\AppData\Local\Temp)
+LOCKFILE = os.path.join(tempfile.gettempdir(), ".codecarbon.lock")
+
+lock_file_created_by_this_process = False
+
+
+def acquire_lock():
+    """Acquires a lock to ensure only one instance of codecarbon is running."""
+    try:
+        _create_lock_file()
+    except FileExistsError as e:
+        logger.debug("Error:", e)
+        logger.error(
+            "Error: Another instance of codecarbon is already running. Turn off the other instance to be able to run this one. Exiting."
+        )
+        sys.exit(1)
```

Comment:
I think it will kill the process that use CodeCarbon : It may be better to just stop the CodeCarbon scheduler.

Reply:
@benoit-cty  You are right, thanks! Checkout the new commit in #[1d2329d](https://github.com/mlco2/codecarbon/pull/562/commits/1d2329d6886d314741e57ce333fb2e25eabf9507)
I created a class to store the lock and have the 2 methods `lock.aquire` and `lock.release` handled by EmissionsTracker. Tell me what you think